Tests/Remove-JiraIssue.Tests.ps1

```diff
@@ -0,0 +1,272 @@
+                        "name": "Medium",
+                        "id": "4"
+                      },
+                      "issuetype": {
+                        "self": "https://jiraserver.example.com/rest/api/2/issuetype/3",
+                        "id": "3",
+                        "description": "A task that needs to be done.",
+                        "iconUrl": "https://jiraserver.example.com/secure/viewavatar?size=xsmall&avatarId=11018&avatarType=issuetype",
+                        "name": "Task",
+                        "subtask": false,
+                        "avatarId": 11018
+                      }
+                    }
+                  },
+                  "subtasks": [],
+                  "project": {
+                    "self": "https://jiraserver.example.com/rest/api/2/project/14801",
+                    "id": "14801",
+                    "key": "TEST",
+                    "name": "TEST - Service Desk",
+                    "avatarUrls": {
+                      "48x48": "https://jiraserver.example.com/secure/projectavatar?avatarId=12003",
+                      "24x24": "https://jiraserver.example.com/secure/projectavatar?size=small&avatarId=12003",
+                      "16x16": "https://jiraserver.example.com/secure/projectavatar?size=xsmall&avatarId=12003",
+                      "32x32": "https://jiraserver.example.com/secure/projectavatar?size=medium&avatarId=12003"
+                    }
+                  },
+                  "description": "Test sub-task.",
+                  "summary": "Test Sub-Task",
+                  "comment": {
+                    "comments": [],
+                    "maxResults": 0,
+                    "total": 0,
+                    "startAt": 0
+                  }
+                }
+              }
+'@
+
+        }
+
+        Mock Get-JiraConfigServer -ModuleName JiraPS {
+            Write-Output $jiraServer
+        }
+
+        Mock Get-JiraIssue {
+            $obj = $TestIssueJSONs[$Key] | ConvertFrom-Json
+
+            $obj.PSObject.TypeNames.Insert(0, 'JiraPS.Issue')
+
+            $obj | Add-Member -MemberType ScriptMethod -Name ToString -Value {return ""} -Force
+            return $obj
+        }
+
+        Mock Invoke-JiraMethod -ModuleName JiraPS -ParameterFilter {$URI -like "$jiraServer/rest/api/*/issue/TEST-1?*" -and $Method -eq "Delete"} {
+            return $null
+        }
+
+        Mock Invoke-JiraMethod -ModuleName JiraPS -ParameterFilter {$URI -like "$jiraServer/rest/api/*/issue/TEST-2?deleteSubTasks=False" -and $Method -eq "Delete"} {
+            $MockedResponse = @"
+            {
+                "errorMessages": [
+                  "The issue 'TEST-2' has subtasks.  You must specify the 'deleteSubtasks' parameter to delete this issue and all its subtasks."
+                ],
+                "errors": {}
+              }
+"@ | ConvertFrom-Json
+            Resolve-JiraError $MockedResponse -WriteError
```

Comment:
You are not supposed to use another function from the module in the UnitTests.
Please use Write-Error instead

Reply:
Resolved with commit 575b88837704cd78a78e53ade5c264639a0af5de